### DISCUSSION TITLE
What should be the Arcus strategy on targetting .NET frameworks ?

### DISCUSSION THREAD

**Discussion by fgheysels** [arcus-azure/arcus#180] [General]
We're currently supporting (or moving to supporting) .NET Core 3.x, .NET 6 and .NET Standard 2.1 in all our Arcus projects.

I wonder if we should explicitly still support .netstandard 2.1.
If we look at [this overview](https://docs.microsoft.com/en-us/dotnet/standard/net-standard#net-implementation-support), it seems like supporting .netstandard 2.1 next to .NET 6 / .NET Core doesn't really make much sense, or am I mistaken ?

By migrating from .netstandard2.0 to .netstandard2.1, we no longer support .NET Framework.
Question that remains: will we support mono & Xamarin ?

Some additional reading
https://devblogs.microsoft.com/dotnet/the-future-of-net-standard/#what-you-should-target

Any ideas @stijnmoreels  @pim-simons  @ALaks96 @gverstraete  ?

**Comment by stijnmoreels** [comment]
With .netstandard, we still have some .NET Core support. I think that was also the reason why we still include this. But, as I recall, .NET Core will be deprecated end of this year, so it's indeed not a very long timespan or added-value that it would have.

We could consider moving directly to .NET 6, bc the Arcus libraries will otherwise have two major 'breaking changes' (.NET Core > .NET standard, .NET standard > .NET 6). And it takes quite some work to migrate every repo, and dependent repo towards this.

**Reply by gverstraete** [reply]
But .NET 6 & .NET Core 3.1 are not compatible. I don't it's a very good approach to force migrations... I would try to support 3.1 & 6 for now...

**Reply by stijnmoreels** [reply]
Yes, that's how we do it now. 😉

**Reply by fgheysels** [reply]
My question is rather: is it necessary to explicitly target .netstandard2.1 ?
.NET 6 implements .Netstandard2.1 and .NET Core afaik does that as well.

Should we also announce that .netstandard 2.0 will no longer be supported as from version x ?

**Reply by gverstraete** [reply]
With .Netstandard2.1 we provide to the customer some more support out of the box. So, I would personally go for that. .netstandard 2.0 is already by it's using out of support... so we can remove that one imo.

**Reply by stijnmoreels** [reply]
What do we do with projects that are only targeting .netstandard2.1? Should they also target .NET 6? Since, we want to migrate towards that.
